### PR TITLE
fix: Select input name instead of file in `ParseFile`

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue
Select the file name from the input name instead of file

Closes: #1011

## Approach
n/a

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Comment by mbfakourii** [comment]
@mtrezza 

The two failed items are not related to the codes of this PR. I will check these two items in another PR

**Comment by mtrezza** [comment]
Could you add a test for this?

**Review by mtrezza — COMMENTED** [review]
Could you add a test for this?

**Comment by mbfakourii** [comment]
> Could you add a test for this?

Added.

File: packages/dart/test/src/objects/parse_file_test.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  group('Parse X File', () {
+    test('should return a correct name', () {
+      // arrange
+      File file = File('/sdcard/aa/aa.jpg');
+
+      // act
+      final parseFile = ParseFile(file, name: 'bb.jpg');
+
+      // assert
+      expect(parseFile.name, 'bb.jpg');
```
Comment (mtrezza) [inline packages/dart/test/src/objects/parse_file_test.dart:21]:
```suggestion
      File file = File('/sdcard/aa/aa.jpg');
      final parseFile = ParseFile(file, name: 'bb.jpg');
      expect(parseFile.name, 'bb.jpg');
```

File: packages/dart/test/src/objects/parse_x_file_test.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  group('Parse X File', () {
+    test('should return a correct name', () {
+      // arrange
+      XFile file = XFile('/sdcard/aa/aa.jpg');
+
+      // act
+      final parseFile = ParseXFile(file, name: 'bb.jpg');
+
+      // assert
+      expect(parseFile.name, 'bb.jpg');
```
Comment (mtrezza) [inline packages/dart/test/src/objects/parse_x_file_test.dart:21]:
```suggestion
      XFile file = XFile('/sdcard/aa/aa.jpg');
      final parseFile = ParseXFile(file, name: 'bb.jpg');
      expect(parseFile.name, 'bb.jpg');
```

**Comment by mtrezza** [comment]
It seems that codecov still reported the new code as not covered. Is there a test case missing?

**Comment by mbfakourii** [comment]
> It seems that codecov still reported the new code as not covered. Is there a test case missing?
> 
> https://github.com/parse-community/Parse-SDK-Flutter/pull/1012/checks?check_run_id=31443404272

No, there is no problem. I have written a test for exactly the same line that had problem. Codecov seems to have not detected well !

**Comment by mbfakourii** [comment]
@mtrezza 

Why are the comments removed? These comments are based on the project's test writing standard

**Comment by mtrezza** [comment]
I don't think these comments add a y value, nor are they currently consistently used, nor are they practical.

If you look at 

https://github.com/parse-community/Parse-SDK-Flutter/blob/master/packages/dart/test/src/objects/parse_base_test.dart

you can see that they are all over the place, sometimes empty, sometimes missing, sometimes they are mixed with other comments with seems confusing.

I'd remove them over time and avoid them for new tests.

**Comment by mbfakourii** [comment]
> > It seems that codecov still reported the new code as not covered. Is there a test case missing?
> > 
> > https://github.com/parse-community/Parse-SDK-Flutter/pull/1012/checks?check_run_id=31443404272
> 
> No, there is no problem. I have written a test for exactly the same line that had problem. Codecov seems to have not detected well !

@mtrezza

**Comment by mtrezza** [comment]
Alright, let's disregard the coverage for that line.

**Comment by mbfakourii** [comment]
> What should we do about the failing beta in the CI? I think we've added it to keep the codebase compatible for the new release. Is it failing because of a change in this PR, or is that unrelated?

The reason ci fails is a warning linter in the beta version of Dart and Flutter and is not related to this PR.

**Review by mtrezza — COMMENTED** [review]
Great, if you could add the version change then we go ahead and merge.

**Comment by mbfakourii** [comment]
> Great, if you could add the version change then we go ahead and merge.

added

**Review by mtrezza — APPROVED** [review]
Looks good!

**Pull request merged**